lib/sandbox.js

```diff
@@ -82,66 +98,76 @@ Sandbox.prototype.answerTask = function(connection, data) {
 };
 
 Sandbox.prototype.executeScript = function(connection, data) {
-  var _this = this;
-
-  var contextObject = {
-    runTask: function (taskName, options) {
+  let contextObject = {
+    runTask: (taskName, options) => {
       try {
-        connection.write(EJSON.stringify({
+        const pack = Buffer.from(EJSON.stringify({
           task: taskName,
           options: options || {}
-        }) + '\u0000'); // task seperator
-      } catch(e) {
-        _this._sendError(connection, e, false);
+        }));
+        const timeStamp = this.nanoTime();
+        this.mmapObject[`type${timeStamp}`] = 'task';
+        this.mmapObject[`sandTask${timeStamp}`] = pack;
```

Comment:
deleting these is only possible from a read completion callback from the master/parent which should still be considered

Reply:
true, but we have to delete them otherwise this mmapObject will slowly grow as the child process is used over and over for different executions, until it explodes